internal/datastore/blockstore.go

```diff
@@ -45,6 +44,22 @@ import (
 // respective substores don't need to optimize or worry about Batching/Txn.
 // Hence the simplified ReaderWriter.
 
+func newCIDKeyBytes(c cid.Cid) ([]byte, error) {
+	sep := []byte("/")
+	b := make([]byte, 0, c.ByteLen()+len(sep))
+	buf := bytes.NewBuffer(b)
+	_, err := buf.Write(sep)
```

Comment:
in what circumstance would this fail? and how would that be different from the `WriteBytes()` call a few lines later?